pkg/kubefnord/join.go

```diff
@@ -262,8 +205,12 @@ func (j *joinFederation) Run(cmdOut io.Writer, fedClientsets util.FedClientset,
 	// Create a service account and use its credentials.
 	glog.V(2).Info("Creating cluster credentials secret")
 
+	if secretName == "" {
+		secretName = names.SimpleNameGenerator.GenerateName(joiningClusterName + "-")
```

Comment:
I think it would be preferable to use the GenerateName field instead of pre-generating the name.  This would ensure secret creation regardless of existing secrets, since the system will ensure a unique name.